lib/realtime/trip_matcher.ex

```diff
@@ -119,4 +124,43 @@ defmodule Realtime.TripMatcher do
       {key, value} -> "#{key}=#{inspect(value)}"
     end)}"
   end
+
+  @spec populate_actual_departures([Vehicle.t()], DateTime.t()) :: [Vehicle.t()]
+  def populate_actual_departures(vehicles, now) do
+    service_date = Util.Time.service_date_for_utc_datetime(now)
+
+    search_cutoff = DateTime.add(now, @event_search_cutoff_m, :minute)
+    search_stations = Enum.map(VehicleEventDetector.terminals(), &elem(&1, 0))
+
+    # Get relevant events from the database
+    events =
+      Repo.all(
+        from(event in VehicleEvent,
+          where:
+            event.service_date == ^service_date and event.arrival_departure == :departure and
+              event.station_id in ^search_stations and event.timestamp > ^search_cutoff,
```

Comment:
It occurs to me I could change this where clause to be the full set of origins in `vehicles`, since we have that so easily accessible. That way we pick up expresses and other moves. I'll adjust tomorrow